libs/base/Syntax/PreorderReasoning.idr

```diff
@@ -22,9 +22,12 @@ data FastDerivation : (x : a) -> (y : b) -> Type where
   (~~) : FastDerivation x y -> (step : Step y z) -> FastDerivation x z
 
 public export
-Calc : {0 x : a} -> {0 y : b} -> FastDerivation x y -> x ~=~ y
-Calc (|~ x) = Refl
-Calc ((~~) der (_ ...(Refl))) = Calc der
+Calc : {0 x : a} -> {0 y : b} -> (0 der : FastDerivation x y) -> x ~=~ y
```

Comment:
For anyone watching at home, the key thing to notice here is
the `0`  in `(0 der : ...)` .